### ISSUE TITLE
Feature Need : Automatic support completions of  a bash file 

### ISSUE THREAD

**Issue by ralic** [scop/bash-completion#175]
Is it possible to automatic support completions of  a bash file, to turn all function written inside for auto-completable ?  For example ,  .bash_profile or .bashrc

**Comment by scop** [comment]
I'm sorry, I don't think I understand the question. Could you rephrase?

**Comment by ralic** [comment]
Let me use two example to help us understand this question .
1) In linux shell, when we press control-R , we can find the command we entered in the past.  (Reverse search)

2) When we type "set | grep '()' , all the function written in a bashrc or bash_profile ... or any function that have been loaded in the bash shell, shall be considered as entered as well.  So when we press "Tab", the bash auto-completion may also guess the extra command the user defined earlier in the shell. 

For example : 
```
brewbot () 
brewbottles () 
brewcc () 
brewpush () 
brewtree () 
bsc () 
cadnd () 
calc () 
capk () 
capp () 
ccc () 
ccmake () 
ccpu () 
cd () 
cdf () 
cefi () 
cgpu () 
checkjdk () 
cheditor () 
cicon () 
cios () 
cip () 
cleantxt () 
cleanyum () 
```

**Comment by barlik** [comment]
@ralic I'm not sure I understand what you mean but function auto-completion already works.

If I define a function in my .bashrc:
`func123() { echo 123; }`

And then I type in: `func1<tab>` in the interactive shell, the bash will auto-complete the function name. And the same holds true for interactively defined functions.

**Comment by ralic** [comment]
It does work, I think there's some setting not working well in the beginning.
